### PR TITLE
Add empty changelog

### PR DESCRIPTION
**Pull request by jisantuc**
This PR adds an empty changelog, since we'll probably want extensions to have those.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
```
Comment (m-mohr) [inline CHANGELOG.md:7]: Give people a bit more info how to structure it, fix the "link":

```suggestion
## [Unreleased]

### Added

### Changed

### Deprecated

### Removed

### Fixed

[Unreleased]: https://github.com/stac-extensions/template/compare/v1.0.0...HEAD
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Changed
+
+### Deprecated
+
+### Removed
+
+### Fixed
+
+[Unreleased]: https://github.com/stac-extensions/template/compare/v1.0.0...HEAD
```
Comment (m-mohr) [inline CHANGELOG.md:19]:
```suggestion
[Unreleased]: <https://github.com/stac-extensions/template/compare/v1.0.0...HEAD>
```

File: README.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 - **Scope:** Item, Collection
 - **Extension [Maturity Classification](https://github.com/radiantearth/stac-spec/tree/master/extensions/README.md#extension-maturity):** Proposal
 - **Owner**: @your-gh-handles @person2
+- [**Changelog**](./CHANGELOG.md)
```
Comment (m-mohr) [inline README.md:9]: I think I'd rather put it to the list below, under JSON Schemas.

**Comment by m-mohr** [comment]
image

File: README.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ This is the place to add a short introduction.
   - [Item example](examples/item.json): Shows the basic usage of the extension in a STAC Item
   - [Collection example](examples/collection.json): Shows the basic usage of the extension in a STAC Collection
 - [JSON Schema](json-schema/schema.json)
+- [**Changelog**](./CHANGELOG.md)
```
Comment (m-mohr) [inline README.md:16]:
```suggestion
- [Changelog](./CHANGELOG.md)
```

**Comment by jisantuc** [comment]
```
-- STAC Version: 1.0.0-rc.1
--- collection: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/collection-spec/json-schema/collection.json 
HTTP ERROR 404
--- collection-assets: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/extensions/collection-assets/json-schema/schema.json 
HTTP ERROR 404
--- item-assets: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/extensions/item-assets/json-schema/schema.json 
HTTP ERROR 404

```

fair

**Pull request merged**